### PR TITLE
[Tests] Add MPS skip decorator

### PR DESCRIPTION
**Pull request by patrickvonplaten**
PR to add MPS skip decorator

### CODE REVIEW

File: tests/pipelines/stable_diffusion_2/test_stable_diffusion_depth.py [hunk 1]
```diff
@@ -40,16 +40,15 @@
...
+from diffusers.utils.testing_utils import require_torch_gpu, skip_mps
 
 from ...test_pipelines_common import PipelineTesterMixin
 
 
 torch.backends.cuda.matmul.allow_tf32 = False
 
 
-@unittest.skipIf(torch_device == "mps", reason="The depth model does not support MPS yet")
+@skip_mps
```
[the remark this answers is not in the document]
Comment (sayakpaul) [reply tests/pipelines/stable_diffusion_2/test_stable_diffusion_depth.py:51]: Perfect!

**Review by pcuenca — APPROVED** [review]
Thanks a lot!

File: tests/pipelines/stable_diffusion_2/test_stable_diffusion_depth.py [hunk 2]
```diff
@@ -208,7 +205,6 @@ def test_save_load_float16(self):
         max_diff = np.abs(output - output_loaded).max()
         self.assertLess(max_diff, 2e-2, "The output of the fp16 pipeline changed after saving and loading.")
 
-    @unittest.skipIf(torch_device != "cuda", reason="float16 requires CUDA")
     def test_float16_inference(self):
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion_2/test_stable_diffusion_depth.py:208]:
```suggestion
     @unittest.skipIf(torch_device != "cuda", reason="float16 requires CUDA")
    def test_float16_inference(self):
```

File: tests/pipelines/stable_diffusion_2/test_stable_diffusion_depth.py [hunk 3]
```diff
@@ -176,7 +174,6 @@ def test_save_load_local(self):
         max_diff = np.abs(output - output_loaded).max()
         self.assertLess(max_diff, 1e-4)
 
-    @unittest.skipIf(torch_device != "cuda", reason="float16 requires CUDA")
     def test_save_load_float16(self):
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion_2/test_stable_diffusion_depth.py:177]:
```suggestion
    @unittest.skipIf(torch_device != "cuda", reason="float16 requires CUDA")
    def test_save_load_float16(self):
```

File: tests/pipelines/stable_diffusion_2/test_stable_diffusion_depth.py [hunk 4]
```diff
@@ -228,10 +224,6 @@ def test_float16_inference(self):
         max_diff = np.abs(output - output_fp16).max()
         self.assertLess(max_diff, 1.3e-2, "The outputs of the fp16 and fp32 pipelines are too different.")
 
-    @unittest.skipIf(
-        torch_device != "cuda" or not is_accelerate_available(),
-        reason="CPU offload is only available with CUDA and `accelerate` installed",
-    )
     def test_cpu_offload_forward_pass(self):
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion_2/test_stable_diffusion_depth.py:227]:
```suggestion
    @unittest.skipIf(
        torch_device != "cuda" or not is_accelerate_available(),
        reason="CPU offload is only available with CUDA and `accelerate` installed",
    )
    def test_cpu_offload_forward_pass(self):
```

**Review by williamberman — APPROVED** [review]
perf

**Pull request merged**